File: phys/362/hw/2/Exercise 6/002-6.js
function variance(data){
	var total = 0,
		sqTotal = 0;
	for(var i = 0; i < data.length; i++){
		total += data[i];
		sqTotal += Math.pow(data[i],2);
	}
	return (sqTotal / data.length) - (Math.pow(total / data.length,2));
}
function testVariance(){
	var data = [],
		n = 2;
	while(1){
		for(var m = 1; m < 11; m++)
			data[m - 1] = 1 + (m * Math.pow(10,-n));
		console.log(data);
		v = variance(data);
		// v -= 8 * Math.exp(10,(-2*n)+1);
		console.log(v);
		console.log(8 * Math.exp(10,(-2*n)+1));
		if(v != 2 * Math.pow(10,-2*n))
			break;
		else
			n++;
	}
	return n;
}
console.log(testVariance());

File: phys/362/2/Exercise 6/002-6.py
import sys
import time

def variance(data):
	total = 0
	sqTotal = 0
	for i in data:
		total += i
		sqTotal += i ** 2
	return (sqTotal / 10) - ((total / 10) ** 2)

def testVariance():
	data = [0,0,0,0,0,0,0,0,0,0]
	n = 1
	while 1:
		for m in range(1,11):
			data[m - 1] = 1 + (m / (10 ** n))
		var = variance(data)
		print(var)
		if var != 8.25 / (10 ** ((2 * n) - 1)):
			break
		else:
			n += 1
	return n

print()
start = time.time()
print(testVariance())
print("%.3f seconds" % (time.time() - start))
print()